### PR TITLE
Switch from yasm to nasm

### PR DESCRIPTION
**Pull request by rcombs**
Our friends at x264, libav, and ffmpeg have all moved (or are moving) in this direction. Nasm is maintained, and has better debugging and macro diagnostics.

### CODE REVIEW

File: libass/ass_fontselect.c [hunk 1]
```diff
@@ -625,7 +625,7 @@ static char *select_font(ASS_FontSelector *priv, ASS_Library *library,
                          ASS_FontStream *stream, uint32_t code)
 {
     ASS_FontProvider *default_provider = priv->default_provider;
-    ASS_FontProviderMetaData meta = {0};
+    ASS_FontProviderMetaData meta = { .families = 0 };
```
Comment (wm4) [inline libass/ass_fontselect.c]: `{0}` is the universal initializer in C, and a compiler shouldn't warn against it. If it does, the warning should be disabled.

**Comment by wm4** [comment]
In general LGTM.

**Comment by grigorig** [comment]
Good idea, the last time I had to install yasm on Windows I was already wondering if the project had died completely. LGTM in general as well, although maybe it'd good to get this tested by people on some of the more unusual platforms.

**Comment by astiob** [comment]
Just wondering: why was Yasm preferred originally? Isn’t it in some sense derived from NASM? It feels a bit weird that the world is now moving “back” to NASM. Is it just that Yasm was in some sense better originally but NASM has improved so much since then that it’s better now?

File: libass/ass_fontselect.c (hunk 1 above)
Comment (astiob) [inline libass/ass_fontselect.c]: Relevant GCC bug reports: https://gcc.gnu.org/bugzilla/show_bug.cgi?id=36750 (fixed ages ago), https://gcc.gnu.org/bugzilla/show_bug.cgi?id=53119 (the fix broke, so refixed in 5.0?), https://gcc.gnu.org/bugzilla/show_bug.cgi?id=64709 (the fix had a bug, fixed in 5.0?), https://gcc.gnu.org/bugzilla/show_bug.cgi?id=80454 (open).

Relevant Clang bug report: https://bugs.llvm.org/show_bug.cgi?id=21689 (open)

File: libass/x86/x86inc.asm [hunk 2]
```diff
@@ -35,13 +35,24 @@
 ; to x264-devel@videolan.org .
 
 %ifndef private_prefix
-    %define private_prefix ass
+    %define private_prefix x264
```
Comment (astiob) [inline libass/x86/x86inc.asm]: This doesn’t seem right.

**Comment by wm4** [comment]
At first there was nasm. Then yasm was created (AFAIK new code base by different people), and either before or after that, nasm declined. For a while yasm was the active project that had all new instructions etc., while nasm seemed dead. But then the situation changed - yasm slowed down, and nasm got alive again. Now it's the other way around, and yasm is dead, while nasm has the new stuff. (I know all this from hearsay only, might be not correct.)

File: libass/x86/x86inc.asm (hunk 2 above)
Comment (rcombs) [inline libass/x86/x86inc.asm]: I imported the latest version outright, and decided to define `private_prefix` in `configure.ac` this time, so we don't have to modify x86inc.asm from "upstream".

Comment (astiob) [inline libass/x86/x86inc.asm]: Ah, I missed the fact that it was defined in `configure.ac`. OK then!

**Comment by rcombs** [comment]
Merged the relevant pieces; reopening a new PR for the other stuff in here

**Pull request closed without merging**